google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1alpha2/BQTableSchemaToProtoDescriptorTest.java

```diff
@@ -197,6 +197,87 @@ public void testStructComplex() throws Exception {
+            .setMode(Table.TableFieldSchema.Mode.REPEATED)
+            .setName("tESt_repEATed")
+            .build();
+    final Table.TableFieldSchema optional =
+        Table.TableFieldSchema.newBuilder()
+            .setType(Table.TableFieldSchema.Type.INT64)
+            .setMode(Table.TableFieldSchema.Mode.NULLABLE)
+            .setName("test_opTIONal")
+            .build();
+    final Table.TableFieldSchema test_int =
+        Table.TableFieldSchema.newBuilder()
+            .setType(Table.TableFieldSchema.Type.INT64)
+            .setMode(Table.TableFieldSchema.Mode.NULLABLE)
+            .setName("TEST_INT")
+            .build();
+    final Table.TableFieldSchema test_string =
+        Table.TableFieldSchema.newBuilder()
+            .setType(Table.TableFieldSchema.Type.STRING)
+            .setMode(Table.TableFieldSchema.Mode.REPEATED)
+            .setName("TEST_STRING")
+            .build();
+    final Table.TableFieldSchema test_bytes =
+        Table.TableFieldSchema.newBuilder()
+            .setType(Table.TableFieldSchema.Type.BYTES)
+            .setMode(Table.TableFieldSchema.Mode.REQUIRED)
+            .setName("TEST_BYTES")
+            .build();
+    final Table.TableFieldSchema test_bool =
+        Table.TableFieldSchema.newBuilder()
+            .setType(Table.TableFieldSchema.Type.BOOL)
+            .setMode(Table.TableFieldSchema.Mode.NULLABLE)
+            .setName("TEST_BOOL")
+            .build();
+    final Table.TableFieldSchema test_double =
+        Table.TableFieldSchema.newBuilder()
+            .setType(Table.TableFieldSchema.Type.DOUBLE)
+            .setMode(Table.TableFieldSchema.Mode.REPEATED)
+            .setName("TEST_DOUBLE")
+            .build();
+    final Table.TableFieldSchema test_date =
+        Table.TableFieldSchema.newBuilder()
+            .setType(Table.TableFieldSchema.Type.DATE)
+            .setMode(Table.TableFieldSchema.Mode.REQUIRED)
+            .setName("TEST_DATE")
+            .build();
+    final Table.TableFieldSchema option_test =
+        Table.TableFieldSchema.newBuilder()
+            .setType(Table.TableFieldSchema.Type.STRUCT)
+            .setMode(Table.TableFieldSchema.Mode.REQUIRED)
+            .addFields(0, required)
+            .addFields(1, repeated)
+            .addFields(2, optional)
+            .setName("option_test")
+            .build();
+    final Table.TableSchema tableSchema =
+        Table.TableSchema.newBuilder()
+            .addFields(0, test_int)
+            .addFields(1, test_string)
+            .addFields(2, test_bytes)
+            .addFields(3, test_bool)
+            .addFields(4, test_double)
+            .addFields(5, test_date)
+            .addFields(6, option_test)
+            .build();
+    final Descriptor descriptor =
+        BQTableSchemaToProtoDescriptor.convertBQTableSchemaToProtoDescriptor(tableSchema);
+    isDescriptorEqual(descriptor, CasingComplex.getDescriptor());
+    System.out.println(descriptor.toProto());
```

Comment:
Do you need this?

Reply:
Sorry forgot to remove the debug message.